homeassistant/components/scrape/__init__.py

```diff
@@ -1 +1,135 @@
 """The scrape component."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import httpx
+import voluptuous as vol
+
+from homeassistant.components.rest.data import RestData
+from homeassistant.components.sensor import (
+    CONF_STATE_CLASS,
+    DEVICE_CLASSES_SCHEMA,
+    STATE_CLASSES_SCHEMA,
+)
+from homeassistant.const import (
+    CONF_ATTRIBUTE,
+    CONF_AUTHENTICATION,
+    CONF_DEVICE_CLASS,
+    CONF_HEADERS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_RESOURCE,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIQUE_ID,
+    CONF_UNIT_OF_MEASUREMENT,
+    CONF_USERNAME,
+    CONF_VALUE_TEMPLATE,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_INDEX, CONF_SELECT, DEFAULT_NAME, DEFAULT_VERIFY_SSL, DOMAIN
+from .coordinator import ScrapeCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+SCRAPE_CONFIG = vol.Schema(
+    {
+        vol.Required(CONF_RESOURCE): cv.string,
```

Comment:
You should import RESOURCE_SCHEMA from `homeassistant.components.rest.schema` and use it here.
Then you have automatic access to: `CONF_RESOURCE`, `CONF_AUTHENTICATION`, `CONF_HEADERS`, `CONF_USERNAME`, `CONF_PASSWORD`, `CONF_VERIFY_SSL`
And it adds `CONF_RESOURCE_TEMPLATE`, `CONF_PARAMS`, `CONF_METHOD`, `CONF_PAYLOAD`, `CONF_TIMEOUT`

```suggestion
        **RESOURCE_SCHEMA,
```